### ISSUE TITLE
Non-deterministic test: RxRingBufferWithoutUnsafeTest.testConcurrency

### ISSUE THREAD

**Issue by davidmoten** [ReactiveX/RxJava#2908]
Concurrency failure on RxRingBuffer doesn't look good. Hopefully a problem with the test only!
```
rx.internal.util.RxRingBufferWithoutUnsafeTest > testConcurrency FAILED
    java.lang.AssertionError: expected:<6920> but was:<6808>
        at org.junit.Assert.fail(Assert.java:93)
        at org.junit.Assert.failNotEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:128)
        at org.junit.Assert.assertEquals(Assert.java:472)
        at org.junit.Assert.assertEquals(Assert.java:456)
        at rx.internal.util.RxRingBufferWithoutUnsafeTest.testConcurrency(RxRingBufferWithoutUnsafeTest.java:139)
```

**Comment by akarnokd** [comment]
Found the bug, see PR #2909.

**Comment by akarnokd** [comment]
Closed via #2909
